allensdk/brain_observatory/behavior/behavior_metadata.py

```diff
@@ -139,3 +147,220 @@ def get_task_parameters(data: Dict) -> Dict:
     task_parameters['n_stimulus_frames'] = n_stimulus_frames
 
     return task_parameters
+
+
+class BehaviorMetadata:
+    """Container class for behavior metadata"""
+    def __init__(self, extractor: Any,
+                 stimulus_timestamps: np.ndarray,
+                 behavior_stimulus_file: pd.DataFrame):
+
+        """Note: cannot properly type extractor due to circular dependency
+        between extractor and transformer.
+        TODO fix circular dependency and update type"""
+
+        self._extractor = extractor
+        self._stimulus_timestamps = stimulus_timestamps
+        self._behavior_stimulus_file = behavior_stimulus_file
+        self._exclude_from_equals = set()
+
+    @property
+    def equipment_name(self) -> str:
+        return self._extractor.get_equipment_name()
+
+    @property
+    def sex(self) -> str:
+        return self._extractor.get_sex()
+
+    @property
+    def age_in_days(self) -> str:
+        return self._extractor.get_age_in_days()
+
+    @property
+    def stimulus_frame_rate(self) -> float:
+        return self._get_frame_rate(timestamps=self._stimulus_timestamps)
+
+    @property
+    def session_type(self) -> str:
+        return self._extractor.get_stimulus_name()
+
+    @property
+    def date_of_acquisition(self) -> datetime:
+        """Return the timestamp for when experiment was started in UTC
+
+        NOTE: This method will only get acquisition datetime from
+        extractor (data from LIMS) methods. As a sanity check,
+        it will also read the acquisition datetime from the behavior stimulus
+        (*.pkl) file and raise a warning if the date differs too much from the
+        datetime obtained from the behavior stimulus (*.pkl) file.
+
+        :rtype: datetime
+        """
+        extractor_acq_date = self._extractor.get_date_of_acquisition()
+
+        pkl_data = self._behavior_stimulus_file
+        pkl_raw_acq_date = pkl_data["start_time"]
+        if isinstance(pkl_raw_acq_date, datetime):
+            pkl_acq_date = pytz.utc.localize(pkl_raw_acq_date)
+
+        elif isinstance(pkl_raw_acq_date, (int, float)):
+            # We are dealing with an older pkl file where the acq time is
+            # stored as a Unix style timestamp string
+            parsed_pkl_acq_date = datetime.fromtimestamp(pkl_raw_acq_date)
+            pkl_acq_date = pytz.utc.localize(parsed_pkl_acq_date)
+        else:
+            pkl_acq_date = None
+            warnings.warn(
+                "Could not parse the acquisition datetime "
+                f"({pkl_raw_acq_date}) found in the following stimulus *.pkl: "
+                f"{self._extractor.get_behavior_stimulus_file()}"
+            )
+
+        if pkl_acq_date:
+            acq_start_diff = (
+                    extractor_acq_date - pkl_acq_date).total_seconds()
+            # If acquisition dates differ by more than an hour
+            if abs(acq_start_diff) > 360:
```

Comment:
This should be 3600 if you want to check for "off by more than an hour"?

Reply:
I can't take credit for this method (was copied over). But good catch